src/engine/strat_engine/pool/v2.rs

```diff
@@ -265,7 +267,8 @@ impl StratPool {
         };
 
         // TODO: Remove in stratisd 4.0
-        let mut needs_save = metadata.thinpool_dev.fs_limit.is_none()
+        let mut needs_save = remove_cache
```

Comment:
Put the ```remove_cache``` part of the expression elsewhere since it will need to stay in stratisd 4.0 (or move the comment).

Reply:
Note that this writes out the metadata even if the pool has no cache. I'm fine with that...just mentioning it.